### PR TITLE
[release-1.6] 🌱 Refactor how tools are built to align with CAPI and get rid of tools go.mod

### PR DESCRIPTION
**Pull request by chrischdi**
Manual cherry-pick of #2080

Remove go module in hack/tools and align to core CAPI with regards to tools installation

* Uses hack/go-install.sh to install:
  * setup-envtest
  * conversion-gen
  * go-apidiff
  * ginkgo
  * kind
* Uses hack/go-tools-build.sh to install:
  * conversion-verifier
  * release-notes
* Adjusts help display function to core CAPI
* Reorders definitions in Makefile to improve comparability to core CAPI Makefile
* Implements hack/go-tools-build.sh to install binaries from sigs.k8s.io/cluster-api/hack/tools/...
* Removes Makefile and go module in hack/tools

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2060

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by chrischdi** [comment]
> @chrischdi: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-cluster-api-provider-vsphere-apidiff-main	[4ec0136](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/commit/4ec013672beffb2c8b054f11cdda1114724aa8b4)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-vsphere/2095/pull-cluster-api-provider-vsphere-apidiff-main/1684207977466171392)	false	`/test pull-cluster-api-provider-vsphere-apidiff-main`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes-sigs&repo=cluster-api-provider-vsphere&pr=2095). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Achrischdi). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues?q=is:issue+is:open) when you hit one in your PR.

Curious where this comes from. Its neither changed by this MR nor the go-apidiff version was changed 🤔

**Comment by chrischdi** [comment]
/hold

to bump go-apidiff

**Comment by chrischdi** [comment]
/hold cancel

Bumped go-apidiff to the same as release-1.7 and main

File: Makefile [hunk 1]
```diff
@@ -146,7 +187,7 @@ LDFLAGS := $(shell hack/version.sh)
 ## --------------------------------------
 
 help: ## Display this help
-	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[0-9A-Za-z_-]+:.*?##/ { printf "  \033[36m%-50s\033[0m %s\n", $$1, $$2 } /^\$$\([0-9A-Za-z_-]+\):.*?##/ { gsub("_","-", $$1); printf "  \033[36m%-50s\033[0m %s\n", tolower(substr($$1, 3, length($$1)-7)), $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
 ## --------------------------------------
 ## Testing
```
Comment (sbueringer) [inline Makefile:193]: Let's drop lint-go in l.200 doesn't make much sense to have it only on old branches

**Pull request merged**